src/clr-angular/data/datagrid/render/cell-renderer.ts

```diff
@@ -22,19 +39,36 @@ export class DatagridCellRenderer implements OnDestroy {
   private subscriptions: Subscription[] = [];
   ngOnDestroy() {
     this.subscriptions.forEach(sub => sub.unsubscribe());
+    if (this.stateSubscription) {
+      this.stateSubscription.unsubscribe();
+    }
+  }
+
+  private stateChanges(state: DatagridColumnState) {
+    if (state.actions && state.actions.length) {
+      state.actions.forEach(action => {
+        switch (action) {
+          case DatagridColumnChanges.WIDTH:
+            this.setWidth(state);
+            break;
+          default:
+            break;
+        }
+      });
+    }
   }
 
   private clearWidth() {
     this.renderer.removeClass(this.el.nativeElement, STRICT_WIDTH_CLASS);
     this.renderer.setStyle(this.el.nativeElement, 'width', null);
   }
 
-  public setWidth(strict: boolean, value: number) {
-    if (strict) {
+  private setWidth(state) {
+    if (state.strictWidth) {
       this.renderer.addClass(this.el.nativeElement, STRICT_WIDTH_CLASS);
     } else {
       this.renderer.removeClass(this.el.nativeElement, STRICT_WIDTH_CLASS);
     }
-    this.renderer.setStyle(this.el.nativeElement, 'width', value + 'px');
+    this.renderer.setStyle(this.el.nativeElement, 'width', state.width + 'px');
```

Comment:
In your new version, you're using a number `strictWidth` that overrides the `width` of the column state. Yet you're still using this logic as if `strictWidth` was a boolean, ignoring the actual number value and still setting the `width` value.

I would go back to a boolean because I think it's clearer, but right now you're in between and it's very confusing.

Reply:
Ah this is a tricky one. The cell previously had a `strictWidth` property that got populated as well as a `strict` boolean. I was trying to condense two properties into one and move that up into the state instead. I haven't pulled this through yet fully (was trying to limit changes) but can work on that too.